File: lib/models/company.dart
class Company {
  String id;
  String companyname;
  String companywebsite;
  String address;
  String dedicatedfield;
  String password;

 Company({
    this.id="1",
    required this.companyname,
    required this.companywebsite,
    required this.address,
    required this.dedicatedfield,
  this.password="1",
  });
  // Company.fromJson(Map json)
  //     : companyname = json['company_name'],
  //       companywebsite = json['Company_website'],
  //       address = json['Address'],
  //       dedicatedfield = json['dedicated_field'],
  //       password = json['password']
  //       ;

  // Map toJson() {
  //   return {
  //     'company_name': companyname,
  //     'Company_website': companywebsite,
  //     'dedicated_field': dedicatedfield,
  //     'Address': address,
  //     'password':password,
  //   };
  // }

   factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
       id : json['_id'],
        companyname : json['company_name'],
        companywebsite :json['Company_website'],
        address : json['Address'],
        password:json['password'],
        dedicatedfield : json['dedicated_field']);
  }
}



 

File: Server/models/company.js
const mongoose = require('mongoose')
const Joi = require('joi')
const jwt = require('jsonwebtoken')
const config = require('config')


const companyschema = mongoose.Schema({
    company_name:{
        type:String,
        required:true,
        minlength:3,
    },
    password:{
        type:String,
        minlength:4,
        required:true,
        maxlength:1024
    },
    Company_website:{
        type:String,
        
    },
    Address:{
        type:String,
        required:true

    },
    dedicated_field:{
        type:String,
        required:true
    },
    role:{
        type:String,
        default:"Company"
    }

})

companyschema.methods.generateregtoken =function(){
const token = jwt.sign({_id:this._id,role:this.role,company_name:this.company_name},config.get('jwtwebtoken'))
return token
}

const Company = mongoose.model('Company',companyschema)

function validateCompany(company){
    const schema = Joi.object({
        // name:Joi.string(),
        password:Joi.string().min(4).max(1024).required(),
        company_name:Joi.string().min(3).required(),
        Company_website:Joi.string(),
        dedicated_field:Joi.string().required(),
        Address:Joi.string().required()
        
    })
    return schema.validate(company)
}
exports.Company=Company
exports.companyschema=companyschema
exports.validateCompany=validateCompany
